File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const n = speeds.length;
    const successes = [];
    
    for (let i = 0; i < n; i++) {
        const progress = progresses[i];
        const speed = speeds[i];
        const day = Math.ceil((100 - progress) / speed);
        successes.push(day);
    }
    
    const answer = [];
    let fst = successes[0];
    let cnt = 1;
    
    for (let i = 1; i < n; i++) {
        if (successes[i] <= fst) {
            cnt += 1;
        } else {
            answer.push(cnt);
            fst = successes[i];
            cnt = 1;
        }
    }
    
    answer.push(cnt);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    # (100 - progress) // speed 만큼 걸림
    # 날짜도 카운트 해야됨
    # 그 뒤 그걸 어딘가에 담고 (며칠 뒤에 되는지 +i + 저 위의 day 만큼 더해서)
    
    n = len(speeds)
    successes = []
    
    for i in range(n):
        progress = progresses[i]
        speed = speeds[i]
        
        day = (100 - progress + speed - 1) // speed
        successes.append(day)
        
    answer = []
    fst = successes[0]
    cnt = 1
    
    for i in range(1, n):
        if successes[i] <= fst:
            cnt += 1
        else:
            answer.append(cnt)
            fst = successes[i]
            cnt = 1
    
    answer.append(cnt)
    return answer
    